parsons/databases/redshift/redshift.py

```diff
@@ -762,6 +770,8 @@ def upsert(self, table_obj, target_table, primary_key, vacuum=True, distinct_che
                     copy_args = dict(copy_args, compupdate=False)
                 self.copy(table_obj, staging_tbl,
                           template_table=target_table,
+                          distkey=primary_key,
```

Comment:
`primary_key` can be a list, but Redshift only allows for a single `distkey`. You could just pick one if there's more than one.. so `primary_keys[0]` (`primary_keys` is generated in line 738 and will always be a list)